### ISSUE TITLE
How do I manually open the same room list from matrix-client-frame?

### ISSUE THREAD

**Issue by Soundtoxin** [alphapapa/matrix-client.el#79]
I'm trying to open the nice-looking roomlist in my current emacs window and I can't see to figure out a command to open that room list. The best I got was to use ace-window to move it from one frame to the other, but now I'm trying to make a new custom layout for matrix so I'm back to square 1.

**Comment by alphapapa** [comment]
There are two nice-looking room lists.  But if you mean the sidebar, it is not currently possible to do so.

**Comment by alphapapa** [comment]
The sidebar is a buffer.  I don't know what you mean by "a normal buffer."  It requires special tracking and handling in order to keep it updated when Matrix events arrive.  The current implementation tracks it by frame using the `frame-purpose` library.  It would require reimplementation in order to change that.  I'm not likely to rewrite that code anytime soon, but patches are welcome.  And as neat as EXWM is, I can't offer specific support for it, sorry.

**Comment by Stebalien** [comment]
Well, yes. Technically everything in emacs is a buffer. However, I can't just call something like `(matrix-client-sidebar)` to create and/or visit the sidebar buffer.

I understand if you don't have time to work on this, but if left-open as a "help wanted" issue, someone might have a time to pick it up.

Note: This isn't just an EXWM specific issue, that's just _my_ motivation. It's an issue for anyone that wants to decide where to put the sidebar.

**Comment by alphapapa** [comment]
> I can't just call something like (matrix-client-sidebar) to create and/or visit the sidebar buffer.

That's true.  This is the part that would require extensive refactoring, see function `matrix-client-frame-update-sidebar`.  It's all made to work with the `frame-purpose` frame.

> I understand if you don't have time to work on this, but if left-open as a "help wanted" issue, someone might have a time to pick it up.

Sure, good idea.  Thanks.

**Comment by divansantana** [comment]
Did anyone find a way to get this package to work well with exwm? I'm struggling to get it to work decently.

**Comment by alphapapa** [comment]
@divansantana You can use it without the room sidebar, which should work fine with exwm.

**Comment by alphapapa** [comment]
Closing issues and archiving repository.  Please see https://github.com/alphapapa/ement.el for the new client.
